### PR TITLE
Allow fully qualified multipart content-type header

### PR DESCRIPTION
**Pull request by simov**
Fixes https://github.com/request/request/issues/1416

The problem here was with poorly written logic about the boundary field in multipart/related content-type header. Now we allow fully qualified multipart headers, including ones containing their own boundary, in that case the boundary property is set from the content-header for that instance (see test-multipart.js)

I've done some major refactoring on the chunked detection tests and moved them to test-multipart-chunked.js I also fixed the logic where the transfer-encoding header overrides the value set through the chunked key, because the header wasn't removed anyway, and I don't think that should be the case either. Probably the previous tests were not catching this edge case.

### CODE REVIEW

**Comment by simov** [comment]
@pronvit can you test out this branch and see if it works for your case?

File: lib/multipart.js [hunk 1]
```diff
@@ -51,11 +51,16 @@ Multipart.prototype.setHeaders = function (chunked) {
...
-  var contentType = (!header || header.indexOf('multipart') === -1)
-    ? 'multipart/related'
-    : header.split(';')[0]
 
-  self.request.setHeader('content-type', contentType + '; boundary=' + self.boundary)
+  if (!header || header.indexOf('multipart') === -1) {
+    self.request.setHeader('content-type', 'multipart/related; boundary=' + self.boundary)
+  } else {
+    if (header.indexOf('boundary') !== -1) {
+      self.boundary = header.replace(/.*boundary=([^\s;])+.*/, '$1')
```
Comment (assaf) [inline lib/multipart.js]: `([^\s;])+` picks the first character of the boundary, should be `([^\s;]+)`

Comment (simov) [inline lib/multipart.js]: Thanks for pointing that out @assaf the bug was fixed here https://github.com/request/request/pull/1505 I had a faulty test for that case.
